api/include/opentelemetry/context/context.h

```diff
@@ -0,0 +1,72 @@
+#pragma once
+
+#include "opentelemetry/common/attribute_value.h"
+#include "opentelemetry/context/context_value.h"
+#include "opentelemetry/nostd/string_view.h"
+#include "opentelemetry/trace/key_value_iterable_view.h"
+
+#include <map>
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace context
+{
+
+// The context class provides a context identifier.
+// This is a dummy class that is meant to be overridden,
+// the methods return default values.
+class Context
+{
+
+public:
+  Context() = default;
+
+  // Contructor, creates a context object from a map of keys
+  // and identifiers.
+  template <class T, nostd::enable_if_t<trace::detail::is_key_value_iterable<T>::value> * = nullptr>
+  Context(const T &keys_and_values)
+  {
+    trace::KeyValueIterableView<T> iterable{keys_and_values};
+    iterable.ForEachKeyValue([&](nostd::string_view key, context::ContextValue value) noexcept {
+      context_map_[std::string(key)] = value;
+      return true;
+    });
+  }
+
+  // Accepts a key and a value and then returns a new context that
+  // contains both the original pairs and the new pair.
+  template <class T>
+  Context SetValue(nostd::string_view key, T &value) noexcept
+  {
+    std::map<std::string, context::ContextValue> context_map_copy = context_map_;
+    context_map_copy[std::string(key)]                            = value;
```

Comment:
What would happen if an library (compilation unit A) is making a call to this method, while the context object was constructed by another binary (compilation unit B), while A and B are using different version of STL?

Reply:
I think I solved the issue by putting the map into a KeyValueIterableView, and then copying the data over.